### ISSUE TITLE
aws_iam_policy_document Bool condition always shows diff

### ISSUE THREAD

**Issue by ghost** [hashicorp/terraform-provider-aws#19245]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Relates #21968

### Terraform CLI and Terraform AWS Provider Version

```bash
Terraform v0.14.3
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.38.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

### Affected Resource(s)

* aws_efs_file_system_policy
* aws_iam_policy_document

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
locals {
  aws_efs_file_system_arn = aws_efs_file_system.fs.arn
}

data "aws_iam_policy_document" "efs_policy" {
  version = "2012-10-17"

  statement {
    sid = "Allow mount and write"

    actions = [
      "elasticfilesystem:ClientWrite",
      "elasticfilesystem:ClientRootAccess",
      "elasticfilesystem:ClientMount"
    ]

    resources = [local.aws_efs_file_system_arn]

    principals {
      type        = "AWS"
      identifiers = ["*"]
    }
  }

  statement {
    sid = "Enforce in-transit encryption for all clients"

    effect = "Deny"

    actions = ["*"]

    resources = [local.aws_efs_file_system_arn]

    principals {
      type        = "AWS"
      identifiers = ["*"]
    }

    condition {
      test     = "Bool"
      variable = "aws:SecureTransport"
      values   = ["false"]
    }
  }
}

resource "aws_efs_file_system_policy" "policy" {
  file_system_id = aws_efs_file_system.fs.id

  policy = data.aws_iam_policy_document.efs_policy.json # TODO: always shows change, need to raise a ticket with aws provider
}

resource "aws_efs_file_system" "fs" {
  encrypted  = true
  kms_key_id = var.kms_key_id

  lifecycle_policy {
    transition_to_ia = "AFTER_14_DAYS"
  }

  tags = {
    Name = format("efs-%s-%s-%s-%s-%s", terraform.workspace, "test", "fs", "test", "test")
  }
}
```

### Expected Behavior

Plan should show no changes.

### Actual Behavior

```hcl
Terraform will perform the following actions:

  # module.shared_data_storage.module.efs.aws_efs_file_system_policy.policy will be updated in-place
  ~ resource "aws_efs_file_system_policy" "policy" {
        id             = "fs-123123"
      ~ policy         = jsonencode(
          ~ {
              ~ Statement = [
                    {
                        Action    = [
                            "elasticfilesystem:ClientWrite",
                            "elasticfilesystem:ClientRootAccess",
                            "elasticfilesystem:ClientMount",
                        ]
                        Effect    = "Allow"
                        Principal = {
                            AWS = "*"
                        }
                        Resource  = "arn:aws:elasticfilesystem:us-east-2:123412341234:file-system/fs-123123"
                        Sid       = "Allow mount and write"
                    },
                  ~ {
                      ~ Condition = {
                          ~ Bool = {
                              ~ aws:SecureTransport = "false" -> [
                                  + "false",
                                ]
                            }
                        }
                        # (5 unchanged elements hidden)
                    },
                ]
                # (1 unchanged element hidden)
            }
        )
        # (1 unchanged attribute hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. Run `terraform apply`
2. Run `terraform plan`

### Important Factoids

This issue wasn't there before upgrading to latest aws provider version.

**Comment by anGie44** [comment]
Hi @IrmantasM , thank you for raising this issue. To help backtrack where this may have been introduced, do you mind providing which provider version you were using before upgrading to the latest? thanks in advance!

**Comment by jooh-lee** [comment]
This is not an issue in provider version v3.27.0. It was introduced in v3.28.0

**Comment by IrmantasM** [comment]
Hi @anGie44,  I was using `3.11.0`, but as per @jooh-lee seems like it was introduced just now.

**Comment by anGie44** [comment]
Can confirm with @jooh-lee's comment above that the behavior is present as early as `v.3.28.0` of the provider .

**Comment by samm-git** [comment]
Same here.  Getting such on all conditions, e.g.

```hcl
                      ~ Condition = {
                          ~ Bool         = {
                              ~ aws:SecureTransport = "true" -> [
                                  + "true",
                                ]
                            }
                          ~ StringEquals = {
                              ~ elasticfilesystem:AccessPointArn = "arn:aws:elasticfilesystem:us-east-1:1234567890:access-point/fsap-1234567890" -> [
                                  + "arn:aws:elasticfilesystem:us-east-1:1234567890:access-point/fsap-1234567890",
                                ]
                            }
                        }
```

**Comment by danopia** [comment]
A previous issue (closed with no proper resolution) showing the the StringEquals case: https://github.com/hashicorp/terraform-provider-aws/issues/17623

**Comment by devopsrick** [comment]
Yep this can be mitigated by adding a trailing comma after any item in a list of conditions/resources/actions, including a single item.

**Comment by YakDriver** [comment]
This _should_ be resolved with #22067 (mostly about IAM but included an updated version of `awspolicyequivalence` also). However, until we create a test to ensure it is fixed, I'm going to leave this open.
